pkg/util/loser/tree.go

```diff
@@ -0,0 +1,132 @@
+// Loser tree, from https://en.wikipedia.org/wiki/K-way_merge_algorithm#Tournament_Tree
+
+package loser
+
+type Value interface{}
+
+type Sequence interface {
+	Next() bool // Advances and returns true if there is a value at this new position.
+}
+
+func NewMerge[E Value, S Sequence](nElements int, maxVal E, at func(S) E, less func(E, E) bool, close func(S)) *Tree[E, S] {
```

Comment:
FWIW I put it there to future-proof the interface in case we do want constraints on `Value`.
At present it is only used in the tests in one place, so probably no need to keep it.